### PR TITLE
Add search functionality

### LINKED ISSUE #42: Search Bar Functionality
Blocked by:
- Issue #41 and Issue #56 : Functionality needs to be implemented first

### PR DESCRIPTION
**Pull request by austincho**
**Overview**
-
The changes in this PR connects the backend `/search` endpoint with the redux data store in the front end and allows for the ability to search for positions using the search bar. Resolves #42 and unblocks #43.

We likely want to implement a loading spinner while we wait for the response from the backend. My plan is to work on the loading spinner when I work on the paginations for searches that result in over 9 positions (#43).

### CODE REVIEW

File: client/src/redux/reducers/positionReducer.js [hunk 1]
```diff
@@ -1,22 +1,28 @@
...
   switch (action.type) {
     case GET_COMMON_POSITIONS:
       return {
-        ...state,
+        ...state, 
+        currentPositions: action.payload,
         commonPositions: action.payload,
       }
-    case SET_CURRENT_POSITION:
+    case CURRENT_POSITION:
```
Comment (valarao) [inline client/src/redux/reducers/positionReducer.js]: Needs to be named better as an "action" - probably `SET_CURRENT_POSITIONS`

File: client/src/components/dashboard/SearchBar.js [hunk 2]
```diff
@@ -34,14 +39,35 @@ const useStyles = makeStyles(theme => ({
...
 }));
 
-// TODO: Add functionality to query and filter for openings. 
+
 const SearchBar = () => {
+  const searchFilter = useRef('');
   const classes = useStyles();
+  const dispatch = useDispatch();
+
+  function onKeyDown(e) {
```
Comment (valarao) [inline client/src/components/dashboard/SearchBar.js]: Convert to arrow function for standardization

File: client/src/components/dashboard/SearchBar.js [hunk 3]
```diff
@@ -34,14 +39,35 @@ const useStyles = makeStyles(theme => ({
...
+    switch(e.key) {
+      case 'Enter':
+        onClick();
+        break;
+      default:
+        return;
+    }
+  }
+
+  function onClick() {
```
Comment (valarao) [inline client/src/components/dashboard/SearchBar.js]: Convert to arrow function for standardization

File: client/src/components/dashboard/PositionCard.js [hunk 4]
```diff
@@ -53,14 +54,14 @@ const PositionCard = (props) => {
       <NotationDisplay
         open={open}
         handleClose={handleClose}
-        name={name}
+        name={nameToUse}
```
Comment (valarao) [inline client/src/components/dashboard/PositionCard.js]: Hmm, I thought what we discussed was to only show the base opening name card. And then when we click on the base opening card, we'll show a dropdown of variations or something.

File: client/src/redux/reducers/positionReducer.js [hunk 5]
```diff
@@ -1,22 +1,28 @@
...
       return {
         ...state,
-        currentPosition: action.payload,
-      };
+        currentPositions: action.payload,
+      }
+      case SWITCH_TO_COMMON:
```
Comment (valarao) [inline client/src/redux/reducers/positionReducer.js]: Don't think it makes sense to have this action

File: client/src/redux/reducers/positionReducer.js [hunk 6]
```diff
@@ -1,22 +1,28 @@
...
 
 const initialState = {
+  currentPositions: null,
   commonPositions: null,
-  currentPosition: null,
 };
 
 const positionReducer = (state = initialState, action) => {
   switch (action.type) {
     case GET_COMMON_POSITIONS:
```
Comment (valarao) [inline client/src/redux/reducers/positionReducer.js:9]: I wouldn't consolidate setting `currentPositions` in the same reducer. Rather, just make two dispatches.

**Review by valarao — CHANGES_REQUESTED** [review]
Great work, Austin - left a few comments.

Also, I noted a potential backend issue: When I try to search "defense", nothing shows up (even though a lot of openings start with defense). In addition to addressing my comments, can you double check the searching algorithm? 

Thanks and let me know if you have any questions!

**Comment by austincho** [comment]
> Great work, Austin - left a few comments.
> 
> Also, I noted a potential backend issue: When I try to search "defense", nothing shows up (even though a lot of openings start with defense). In addition to addressing my comments, can you double check the searching algorithm?
> 
> Thanks and let me know if you have any questions!

Hey Miguel, currently when the search happens it looks if the baseOpening.name field or the variation field contains what you searched. I believe the issue that you're running into is because of the fact that the our data uses the "Defence" spelling vs "Defense".

**Review by valarao — APPROVED** [review]
LGTM!

**Review by joyx9 — APPROVED** [review]

**Pull request merged**